tests/bq_test_helpers.py

```diff
@@ -0,0 +1,114 @@
+"""
+Shared helper functions for BigQuery-related tests
+"""
+import datetime
+import warnings
+from collections import OrderedDict
+from typing import Any, List, Union, Dict, Iterable
+
+import mock
+from google.cloud import bigquery
+from google.cloud.bigquery import TableReference
+from google.cloud.bigquery.table import TableListItem
+
+_TYPE_TO_FIELD_TYPE = {
+    str(int): 'INT64',
+    str(str): 'STRING',
+    str(float): 'FLOAT64',
+    str(datetime.date): 'DATE',
+    str(datetime.datetime): 'TIMESTAMP'
+}
+dict_types = Union[Dict, OrderedDict]
```

Comment:
I didn't go through creating a NamedTuple because I like this setup you've got here.  So I created this very small patch file.  What do you think?  I know we could let someone use strings, but if these helper functions were used by an integration test, a string object iterable would blow this stuff up.  Because I find it unlikely that we'll have single character column names in an integration test.  On that subject, do you intend for these functions to be used by integration and unit tests alike?

[non_string_iterables.txt](https://github.com/all-of-us/curation/files/5497459/non_string_iterables.txt)

Reply:
The patch would permit `key_order` to be `List[tuple]` or `List[List]` whereas it is most likely to be passed as a `List[str]` ([example](https://github.com/all-of-us/curation/blob/mv/purge_hpo/tests/unit_tests/data_steward/utils/bq_test.py#L188-L189)) as it should represent the keys within each row (type=dict) in the order that the result schema. A value belonging to any immutable (hashable?) type is permitted to be used as a key in a dictionary.

I thought maybe the patch had a typo and `non_string_iterable_types` should be the container type (?), but my IDE is complaining this is invalid.
![image](https://user-images.githubusercontent.com/8049605/98300299-164b1f00-1f87-11eb-9a55-90f987df29f9.png)

Reply:
There is a lot of debate about how to use [Iterable without using a string](https://github.com/python/typing/issues/256).  I don't think there's been a good answer for how to proceed forward.  I'm dropping this.